util/gen_effect_id.ts

```diff
@@ -0,0 +1,132 @@
+import path from 'path';
+import { CoinachWriter } from './coinach';
+import { Table, cleanName, getIntlTable } from './csv_util';
+
+// Maybe this should be called Status like the table, but everything else
+// says gain/lose effects.
+const effectsOutputFile = 'effect_id.ts';
+
+// TODO: add renaming?
+// Almagest: 563
+
+// There are a looooot of duplicate effect names in pvp, and it's impossible
+// to differentiate other than manually.  There's also older effects that
+// did different things that are still around.
+//
+// This is a map of id to skill name (for smoke testing/documentation).
+const knownMapping = {
+  'Thundercloud': '164',
+  'Battle Litany': '786',
+  'Right Eye': '1910',
+  'Left Eye': '1454',
+  'Meditative Brotherhood': '1182',
+  'Brotherhood': '1185',
+  'Embolden': '1297',
+  'Technical Finish': '1822',
+  'Sheltron': '1856',
+  'Lord of Crowns': '1876',
+  'Lady of Crowns': '1877',
+  'Divination': '1878',
+  'The Balance': '1882',
+  'The Bole': '1883',
+  'The Arrow': '1884',
+  'The Spear': '1885',
+  'The Ewer': '1886',
+  'The Spire': '1887',
+  'Sword Oath': '1902',
+  'Tactician': '1951',
+  // This is for others, 1821 is for self.
+  'Standard Finish': '2105',
+  'The Wanderer\'s Minuet': '2216',
+  'Mage\'s Ballad': '2217',
+  'Army\'s Paeon': '2218',
+  'Stormbite': '1201',
+  'Caustic Bite': '1200',
+  'Windbite': '129',
+  'Venomous Bite': '124',
+  'Flourishing Fan Dance': '1820',
+  'Higanbana': '1228',
+  'Wildfire': '861',
+  'Chain Stratagem': '1221',
+  'Vulnerability Up': '638',
+} as const;
+
+// These custom name of effect will not be checked, but you'd better make it clean.
+// Use this only when you need to handle different effects with a same name.
+const customMapping = {
+  'EmboldenSelf': '1239',
+} as const;
+
+
+const printError = (
+    header: string,
+    what: string,
+    map: Record<string | number, unknown>,
+    key: string,
+) =>
+  console.error(`${header} ${what}: ${JSON.stringify(map[key])}`);
+
+
+const makeEffectMap = (table: Table<'#', 'Name'>) => {
+  const foundNames = new Set();
+
+  const map: { [s: string]: string } = {};
+  for (const [id, effect] of Object.entries(table)) {
+    const rawName = effect['Name'];
+    if (!rawName)
+      continue;
+    const name = cleanName(rawName);
+
+    if (rawName && rawName in knownMapping) {
+      if (id !== knownMapping[rawName as keyof typeof knownMapping]) {
+        printError('skipping', rawName, table, id);
+        continue;
+      }
+    }
+
+    if (name in map) {
+      printError('collision', name, table, id);
+      printError('collision', name, table, map[name] ?? '');
```

Comment:
I think there is a bug with the collision code here, for example `Doom`:
```
collision Doom: {"#":"910","Name":"Doom","Icon":"15503","PartyListPriority":"200"}
collision Doom: {"#":"210","Name":"Doom","Icon":"15503","PartyListPriority":"200"}
collision Doom: {"#":"1769","Name":"Doom","Icon":"15503","PartyListPriority":"200"}
collision Doom: {"#":"1738","Name":"Doom","Icon":"13516","PartyListPriority":"50"}
collision Doom: {"#":"2516","Name":"Doom","Icon":"15503","PartyListPriority":"200"}
collision Doom: {"#":"1970","Name":"Doom","Icon":"15503","PartyListPriority":"200"}
```

But yet when running this script `Doom` appears in the output (where it didn't before).  It should be removed if it has a collision.

Reply:
Transform `map` to a truthy `Map` fixed it.